glib/src/gstring.rs

```diff
@@ -102,6 +102,67 @@ impl GStr {
     pub fn as_c_str(&self) -> &CStr {
         unsafe { CStr::from_bytes_with_nul_unchecked(self.to_bytes_with_nul()) }
     }
+
+    #[doc(alias = "g_utf8_collate")]
+    #[inline]
+    pub fn collate(&self, other: &Self) -> Ordering {
+        unsafe { ffi::g_utf8_collate(self.as_ptr(), other.as_ptr()) }.cmp(&0)
+    }
+
+    #[doc(alias = "g_utf8_casefold")]
+    #[inline]
+    pub fn casefold(&self) -> GString {
```

Comment:
Having these on `&GStr` requires unnecessarily the `\0` terminator to be added. That's not required for the functions here and they would as well work on a normal `&str` (that's the only difference between those two types).

Reply:
Oh yeah I didn't consider that, I guess taking `impl AsRef<str>` would be ideal then? Would these names be okay even if the functions were exposed at the module level?

Reply:
Yeah, maybe as part of the convert module?

Reply:
I decided to keep `collate` as an associated function as it takes null terminated strings, but I moved the rest to `convert`